### PR TITLE
Make automacros checking faster

### PR DESCRIPTION
**Pull request by Henrybk**
Now only automacros with met conditions will be checked on running time on AI_pre hook, and only their timeout will be checked, no more checking conditions during AI_pre.

### CODE REVIEW

File: eventMacro/Core.pm [hunk 1]
```diff
@@ -372,6 +365,75 @@ sub exists_var {
...
 }
 
+sub add_to_triggered_prioritized_automacros_index_list {
+	my ($self, $automacro) = @_;
+	my $priority = $automacro->get_parameter('priority');
+	my $index = $automacro->get_index;
+	
+	push( @{ $self->{triggered_prioritized_automacros_index_list} }, { index => $index, priority => $priority } );
+	
+	my $size = scalar @{ $self->{triggered_prioritized_automacros_index_list} };
```
Comment (allanon) [inline eventMacro/Core.pm:375]: `scalar` is unnecessary here because this is already scalar context.

File: eventMacro/Automacro.pm [hunk 2]
```diff
@@ -33,6 +34,21 @@ sub parse_and_create_conditions {
 	}
 }
 
+sub get_running_status {
```
Comment (allanon) [inline eventMacro/Automacro.pm:37]:
```
sub running_status {
  $_[0]->{running_status} = $_[1] if @_ > 1;
  $_[0]->{running_status};
}
```

File: eventMacro/Core.pm [hunk 3]
```diff
@@ -372,6 +365,75 @@ sub exists_var {
...
+	
+	$self->{automacro_index_to_queue_index}{$index} = $size - 1;
+	
+	my $new_index;
+	
+	if ( $size == 1 ) {
```
Comment (allanon) [inline eventMacro/Core.pm:381]: How about:
```
my $list = $self->{triggered_prioritized_automacros_index_list} ||= [];

# Find where we should insert this item.
my $i;
for ($i = 0 ; $i < @$list && $list->{priority} >= $priority ; $i++) {}

# Insert.
splice @$list, $i, 0, { index => $index, priority => $priority };

# Update indexes.
my $index = $self->{automacro_index_to_queue_index};
$index->{$list->[$_]} = $_ foreach $i .. $#list;

# Return the insertion index.
$i;
```

File: eventMacro/Core.pm (hunk 1 above)
Comment (Henrybk) [reply eventMacro/Core.pm:375]: Forgot I could do that

File: eventMacro/Core.pm (hunk 3 above)
Comment (Henrybk) [reply eventMacro/Core.pm:381]: That seems to be way easier in fact.

**Pull request merged**